speechbrain/core.py

```diff
@@ -878,6 +887,7 @@ def fit_batch(self, batch):
                 if self.check_gradients(loss):
                     self.scaler.step(self.optimizer)
                 self.scaler.update()
+                self.optimizer.zero_grad()
```

Comment:
While I appreciate that this is technically faster than `self.zero_grad()`, I think it would still make more sense to me to call `self.zero_grad()` here. Its more consistent and if for some reason someone wanted to override this function it would be better to call it here as well.

Reply:
I agree, but what about the recipes, which uses custom optimisers and thus has custom `fit_batch`? Should I also call `self.zero_grad()` there?

Reply:
> Nice change! I especially appreciate saving space for evaluation and fixing the gradient accumulation bug. Quick question, have you verified that this does in fact save space for evaluation and there's nothing else (e.g. manual garbage collection) that is needed?

This is what I am evaluating this week, that is why the PR is marked as WIP

Reply:
I don't think we need to add `self.zero_grad()` for recipes with custom `fit_batch`. While it probably wouldn't hurt anything, it doesn't seem a necessary change to me, recipes will continue to work as before. If in the future there is some change to `self.zero_grad()`, the recipes could be updated at that time.

Reply:
Okay, I can confirm that calling `self.zero_grad(set_to_none=True)` will free the memory. So I think you can merge it.